Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+                    //Mols start to have a positive effect on damage after 350
+                    var MoleClamp = Math.Clamp(mixture.TotalMoles / 200f, 0.5f, 1f);
+                    var HeatDamage = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold)*SMcomponent.DynamicHeatResistance);
+                    damage = Math.Max(damage + (Math.Max(MoleClamp * mixture.Temperature - HeatDamage, 0) * SMcomponent.MoleHeatPenalty / 150) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Power only starts affecting damage when it is above 5000
+                    damage = Math.Max(damage + (Math.Max(SMcomponent.Power - SupermatterComponent.PowerPenaltyThreshold, 0f)/500f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Molar count only starts affecting damage when it is above 1800
+                    damage = Math.Max(damage + (Math.Max(mixture.TotalMoles - SupermatterComponent.MolePenaltyThreshold, 0f)/80f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //There might be a way to integrate healing and hurting via heat
+			        //healing damage
+                    if (mixture.TotalMoles < SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        //Only has a net positive effect when the temp is below 313.15, heals up to 2 damage. Psycologists increase this temp min by up to 45
+                        var heatcap = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold) + (45 * SMcomponent.PsyCoeff));
+                        damage = Math.Max(-(2 * (1 - (mixture.Temperature / heatcap))),-2);
+                    }
+
+                    //if there are space tiles next to SM
+                    foreach(var adjacent in _atmosphereSystem.GetAdjacentTileMixtures(Xform.Coordinates))
+                    {
+                        if (adjacent.TotalMoles == 0)
+                        {
+                            if (integrity < 90)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 75)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0009f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 55)
+                                damage = Math.Clamp((SMcomponent.Power * 0.005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 25)
+                                damage = Math.Clamp((SMcomponent.Power * 0.002f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else
+                                break;
```

Comment:
use pattern matching

```csharp
factor = integrity switch
{
    (< 25) => 0.002,
    (<= 55) and (> 25) => 0.005,
    (<= 75) and (> 55) => 0.0009,
    (<= 90) and (> 75) => 0.0005,
    (> 90) => 0,
};

damage = Math.Clamp((SMcomponent.Power * factor) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
```

see https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#relational-patterns for info